src/dsp_tools/commands/id2iri.py

```diff
@@ -61,7 +62,7 @@ def _parse_json_file(json_file: Path) -> dict[str, str]:
 
 def _replace_resptrs(
     tree: etree._Element,
-    mapping: dict[str, str],
+    mapping: Mapping[str, str],
```

Comment:
By doing this, I loosen the constraint on the input, which is okay - or even desirable, because input type annotations should be as loose as possible, while return type annotations should be as strict as possible.

Reply:
why is this a default value?

Reply:
It isn't. The thing with the default value is further below. This here is just a follow-up change

Reply:
Because the data type would not match?
Can we then still give a dict without mypy complaining?

Reply:
Further below, I replaced the mutable dict with the immutable MappingProxyType. But MappingProxyType is not a subtype of dict, so the type has to be annotated with a supertype, "collections.abc.Mapping" in our case. This is the overarching supertype of dict and MappingProxyType.

And yes, of course the actual input doesn't change, we still pass a dict to the method.